tensorflow/contrib/learn/python/learn/datasets/base.py

```diff
@@ -205,11 +205,9 @@ def maybe_download(filename, work_directory, source_url):
     gfile.MakeDirs(work_directory)
   filepath = os.path.join(work_directory, filename)
   if not gfile.Exists(filepath):
-    with tempfile.NamedTemporaryFile() as tmpfile:
-      temp_file_name = tmpfile.name
-      urlretrieve_with_retry(source_url, temp_file_name)
-      gfile.Copy(temp_file_name, filepath)
-      with gfile.GFile(filepath) as f:
-        size = f.size()
-      print('Successfully downloaded', filename, size, 'bytes.')
+    temp_file_name, _ = urlretrieve_with_retry(source_url)
```

Comment:
I've put the filename argument back, with a default of None, similar to urllib.request.urlretrieve() itself. This should allow the tests to remain as-is, while still allowing the fix for the use case in maybe_download()